controller/workitem.go

```diff
@@ -110,6 +111,37 @@ func (c *WorkitemController) List(ctx *app.ListWorkitemContext) error {
 		exp = criteria.And(exp, criteria.Equals(criteria.Field(workitem.SystemState), criteria.Literal(string(*ctx.FilterWorkitemstate))))
 		additionalQuery = append(additionalQuery, "filter[workitemstate]="+*ctx.FilterWorkitemstate)
 	}
+	if ctx.FilterCategory != nil {
+		var relationships []*category.WorkItemTypeCategoryRelationship
+		err := application.Transactional(c.db, func(tx application.Application) error {
+			// Load all workitemtypes related to the specific category
+			relationships, err = tx.Categories().LoadAllRelationshipsOfCategory(ctx, *ctx.FilterCategory)
+			if err != nil {
+				log.Error(ctx, map[string]interface{}{
+					"category_id": *ctx.FilterCategory,
+					"err":         err,
+				}, "failed to list work items in category")
+				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("failed to list work items in category %v", *ctx.FilterCategory)))
+			}
+			return nil
+		})
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("failed to list work items in category %v", *ctx.FilterCategory)))
```

Comment:
@kwk so you tell that this makes sense to you:
```
err := func() err {
   ....
   if err != nil {
				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("failed to list work items in category %v", *ctx.FilterCategory)))
}
})
if err != nil {
				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("failed to list work items in category %v", *ctx.FilterCategory)))
}
```
If the errors are complementaries of coming from different layers, I could understand it. In this case the error is wrapped in the controller and re-wrapped again outside of the caller function. There isn't not different type of errors inside the inner function either.

Moreover if you treat the error as a JSONAPIErrorResponse why to do it twice if there isn't any specialization in the error treatment. To me the following code makes more sense
```
err := func() err {
   ....
   if err != nil {
				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("failed to list work items in category %v", *ctx.FilterCategory)))
}
})
if err != nil {
				return err
}
```